src/mono/mono/metadata/assembly.c

```diff
@@ -1663,27 +1684,36 @@ netcore_load_reference (MonoAssemblyName *aname, MonoAssemblyLoadContext *alc, M
 	 * Try these until one of them succeeds (by returning a non-NULL reference):
 	 * 1. Check if it's already loaded by the ALC.
 	 *
-	 * 2. If it's a non-default ALC, call the Load() method.
+	 * 2. If we have a bundle registered, search the images for a matching name.
+	 *
+	 * 3. If it's a non-default ALC, call the Load() method.
 	 *
-	 * 3. If the ALC is not the default and this is not a satellite request,
+	 * 4. If the ALC is not the default and this is not a satellite request,
 	 *    check if it's already loaded by the default ALC.
 	 *
-	 * 4. If the ALC is the default or this is not a satellite request,
+	 * 5. If the ALC is the default or this is not a satellite request,
 	 *    check the TPA list, APP_PATHS, and ApplicationBase.
 	 *
-	 * 5. If this is a satellite request, call the ALC ResolveSatelliteAssembly method.
+	 * 6. If this is a satellite request, call the ALC ResolveSatelliteAssembly method.
 	 *
-	 * 6. Call the ALC Resolving event.
+	 * 7. Call the ALC Resolving event.
 	 *
-	 * 7. Call the ALC AssemblyResolve event (except for corlib satellite assemblies).
+	 * 8. Call the ALC AssemblyResolve event (except for corlib satellite assemblies).
 	 *
-	 * 8. Return NULL.
+	 * 9. Return NULL.
 	 */
 
 	reference = mono_assembly_loaded_internal (alc, aname, FALSE);
 	if (reference)
 		goto leave;
 
+	if (bundles != NULL) {
+		reference = search_bundle_for_assembly (alc, aname);
+		if (reference) {
```

Comment:
Remove curly braces?

Reply:
Oh never mind, I didn't see commit 2 before making this comment

Reply:
I couldn't be bothered to fix that in the original commit when chunking this up since it's getting squashed anyway :p